pennylane/transforms/sign_expand/sign_expand.py

```diff
@@ -0,0 +1,340 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the sign (and xi) decomposition tape transform, implementation of ideas from arXiv:2207.09479
+"""
+# pylint: disable=protected-access
+import json
+from os import path
+import pennylane as qml
+from pennylane import numpy as np
+from ..batch_transform import batch_transform
+
+
+def ControlledPauliEvolution(theta, wires, pauli_word, controls):
+    r"""Controlled Evolution under generic Pauli words, adapted from the decomposition of
+    qml.PauliRot to suit our needs
+
+
+    Args:
+        theta (float): rotation angle :math:`\theta`
+        pauli_word (string): the Pauli word defining the rotation
+        wires (Iterable, Wires): the wires the operation acts on
+        controls (List[control1, control2]): The two additional controls to implement the
+          Hadamard test and the quantum signal processing part on
+
+    Returns:
+        list[Operator]: decomposition that make up the controlled evolution
+    """
+    active_wires, active_gates = zip(
+        *[(wire, gate) for wire, gate in zip(wires, pauli_word) if gate != "I"]
+    )
+
+    ops = []
+    for wire, gate in zip(active_wires, active_gates):
+        if gate == "X":
+            ops.append(qml.Hadamard(wires=[wire]))
+        elif gate == "Y":
+            ops.append(qml.RX(np.pi / 2, wires=[wire]))
+
+    ops.append(qml.CNOT(wires=[controls[1], wires[0]]))
+    ops.append(qml.ctrl(op=qml.MultiRZ(theta, wires=list(active_wires)), control=controls[0]))
+    ops.append(qml.CNOT(wires=[controls[1], wires[0]]))
+
+    for wire, gate in zip(active_wires, active_gates):
+        if gate == "X":
+            ops.append(qml.Hadamard(wires=[wire]))
+        elif gate == "Y":
+            ops.append(qml.RX(-np.pi / 2, wires=[wire]))
+    return ops
+
+
+def evolve_under(ops, coeffs, time, controls):
+    """
+    Evolves under the given Hamiltonian deconstructed into its Pauli words
+
+    Args:
+        ops (List[Observables]): List of Pauli words that comprise the Hamiltonian
+        coeffs (List[int]): List of the respective coefficients of the Pauliwords of the Hamiltonian
+        time (float): At what time to evaluate these Pauliwords
+    """
+    ops_temp = []
+    for op, coeff in zip(ops, coeffs):
+        pauli_word = qml.grouping.pauli_word_to_string(op)
+        ops_temp.append(
+            ControlledPauliEvolution(
+                coeff * time,
+                wires=op.wires,
+                pauli_word=pauli_word,
+                controls=controls,
+            )
+        )
+    return ops_temp
+
+
+def calculate_Xi_decomposition(hamiltonian):
```

Comment:
Better to stick with python naming convention

```suggestion
def calculate_xi_decomposition(hamiltonian):
```